### PR TITLE
Feat/katy master

### LINKED ISSUE #3: find free hosting solution
@bahmutov - repo is now public and owned by `350-mass-cambridge-somerville` organization. We need find a free hosting + deployment solution.

### LINKED ISSUE #7: enforce consistent style
to avoid mix of tabs vs spaces and inconsistent code

Screen Shot 2020-02-05 at 9 21 31 PM

I think most of the code is using tabs

### LINKED ISSUE #8: card needs white backround
how do we get background colors into the theme?

Screen Shot 2020-02-05 at 9 31 36 PM

The original looks nice

Screen Shot 2020-02-05 at 9 32 25 PM

### LINKED ISSUE #9: How does the "past" work?
I cannot see any requests coming from the application when I click "past", and I would love to recreate the functionality from http://maclimateactions.com/past

### PR DESCRIPTION
**Pull request by ccpowers**
I realized that I had a significant amount of work locally which is why the current site looks different than local dev (sorry!). This:
- fixes / refactors current action view
- moves the login/registration views out of 'pop ups' 
- does some routing work so that back buttons work with the react router urls
- fixes tags display
- updates e2e tests to include past action view, test tags and survey response numbers

I think this is what we should deploy (or some of it), since it is how the site looks now. 

- closes #7
- closes #8 
- closes #9
- closes #3 (Netlify preview, maybe hosting)

### CODE REVIEW

**Comment by bahmutov** [comment]
Screen Shot 2020-02-08 at 11 51 45 AM

we need to make sure to use HTTPS for Ajax requests, where is the IP address coming from? Can we switch to DNS record?

File: netlify.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[build]
+base = "./"
+publish = "build/"
+command = "npm run build"
+
+[build.environment]
+  REACT_APP_BASE_URL = "http://159.89.54.69:8000"
```
Comment (bahmutov) [inline netlify.toml]: this is the API endpoint? we need to use HTTPS domain name

**Comment by bahmutov** [comment]
I have added more tests, including submitting a survey response

I think aside from API http vs https it is good to go

**Comment by ccpowers** [comment]
@bahmutov yeah, i need to update the backend in order to make https work. Am working on that this weekend. Do you know anything about gunicorn/nginx?

**Comment by bahmutov** [comment]
Unfortunately no, one good thing is that during testing we can stub the backend for now

**Comment by bahmutov** [comment]
@ccpowers can you give me access to the Netlify project, please? I want to see if I can configure test run after the deploy

**Comment by ccpowers** [comment]
@bahmutov trying to figure out how to do that...it seems like it will cost $15 a month to have two people access the repo, might need to just use the ten minute heroes email and sign in on that. if you have any idea how to get two people access to a netlify project for free, let me know.

**Comment by bahmutov** [comment]
@ccpowers ok, here is what we will do. People at Netlify are our friends, can you (as the admin on this organization and repo owner) write them a note at https://www.netlify.com/enterprise/contact/

- we are working on climate activism
- our open source repo calling for actions every week is here
- we would love to use Netlify for testing new features and host the main site
- we have two team members currently (write your name and mine) and would love both to work on Netlify settings if needed
- please, pretty please give us an OSS plan, and we will happily post Netlify name in the README and will always mention it during our talks and development presentations

I think this will work!

**Comment by TenMinuteHero** [comment]
@bahmutov Oh cool! I just saw this after deploying using the ten minute heros email address - I emailed you an expiring link with details for that. I'll email the netlify folks and ask for an account, and in the meantime, you can use the tmh account.

**Comment by bahmutov** [comment]
I am reworking how Netlify deploy happens (no longer automatic, but triggered from GH workflow). This will allow us to deploy, then run Cypress tests against the deployed url

**Comment by TenMinuteHero** [comment]
@ccpowers I have changed Netlify site name to be https://maclimateactions.netlify.com from the randomly-assigned name

**Comment by bahmutov** [comment]
@ccpowers I am done with my stuff for this PR, and can help with API?

**Comment by ccpowers** [comment]
@bahmutov yes! i will tag you in my work on the back end explaining where i am

**Pull request merged**